### ISSUE TITLE
Rule proposal: Prefer object form of `addEventListener`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#2067]
### Description

[addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) has a legacy signature where it takes `useCapture` as the third argument. The `options` form should be preferred.

### Fail

```js
addEventListener('click, () => {}, true)
```

### Pass

```js
addEventListener('click, () => {}, {capture: true})
``

### Additional Info

_No response_

**Comment by fisker** [comment]
Name?

- `no-add-event-listerner-legacy-option`
- `no-add-event-listerner-boolean-option`
- `prefer-add-event-listener-object-options`
- `no-boolean-use-capture`

**Comment by silverwind** [comment]
`prefer-add-event-listener-options`?

**Comment by sindresorhus** [comment]
> prefer-add-event-listener-options?

👍

**Comment by sindresorhus** [comment]
Accepted

**Comment by fregante** [comment]
Related: https://ariya.io/2011/08/hall-of-api-shame-boolean-trap

Are there any other APIs with boolean traps in JS? This could use @silverwind’s name or something related to booleans:

- `no-add-event-listener-boolean`
- `no-boolean-traps` (this could be its own typescript rule for function definitions, though)

**Comment by silverwind** [comment]
> Are there any other APIs with boolean traps in JS

Can't think of any but I'm sure there are more such DOM APIs.

> `no-boolean-traps`

Seems nice in theory to have a general rule, but I guess some people prefer to control flagging such stuff on per-case basis, so a general rule might take away such a choice, unless it has options per affected function.
